src/components/MetricGraph.tsx

```diff
@@ -104,22 +114,36 @@ const MetricGraph = ({ streamId, interval, metric }: MetricGraphProps) => {
 
   // Poll graph data
   const graphPollTimeout = useRef<number | undefined>()
-  const graphPoll = useCallback((nextMetric: MetricType) => {
+  const graphPoll = useCallback((nextMetric: MetricType, nextInterval: Interval) => {
     clearTimeout(graphPollTimeout.current)
 
-    const nextValues = (dataRef.current || []).map(({ timestamp, ...rawValue }) => ({
-      x: timestamp,
-      y: rawValue[nextMetric],
-    }))
+    const now = Date.now()
+
+    const nextValues = (dataRef.current || [])
+      .filter(({ timestamp }) => {
+        switch (nextInterval) {
+          case 'realtime':
+            return (now - timestamp) <= REALTIME_WINDOW
+
+          default:
+            break
+        }
+
+        return true
+      })
+      .map(({ timestamp, ...rawValue }) => ({
+        x: timestamp,
+        y: rawValue[nextMetric],
+      }))
     setValues(nextValues)
 
     graphPollTimeout.current = setTimeout(() => {
-      graphPoll(nextMetric)
-    }, 200)
+      graphPoll(nextMetric, nextInterval)
+    }, 500)
```

Comment:
Polling is not the best option but throttling the update function didn't seem to work, didn't debug that now deeper 🤷